pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go

```diff
@@ -120,6 +124,17 @@ func (s *apiScaffolder) Scaffold() error {
 	return nil
 }
 
+func (s *apiScaffolder) updateMainEventRecorder() error {
+	defaultMainPath := "main.go"
+	err := util.InsertCode(defaultMainPath, `Scheme: mgr.GetScheme(),
+	`,
+		fmt.Sprintf(recorderTemplate, strings.ToLower(s.resource.Kind)))
+	if err != nil {
+		return fmt.Errorf("error scaffolding event recorder while creating the controller in main.go: %v", err)
+	}
+	return nil
+}
```

Comment:
@NikhilSharmaWe,
 
Instead of use replace the best approach here would we create a MainUpdater like:

https://github.com/NikhilSharmaWe/kubebuilder/blob/f604a6cb2c4a3ba5425471ace96125630c9a50f3/pkg/plugins/golang/v3/scaffolds/api.go#L106-L110

Then, in this updater we will have another template for the controller in the main: https://github.com/NikhilSharmaWe/kubebuilder/blob/f604a6cb2c4a3ba5425471ace96125630c9a50f3/pkg/plugins/golang/v3/scaffolds/internal/templates/main.go#L100-L106

Could you please add a TODO on top of the method at least such as 

// TODO: replace this implementation by creating its own MainUpdater which will have its own controller template which set the recorder. 

Also, could you please change the target so we ensure that you ONLY set the event.Recorder on the Controller code of the specific api instead of adding the code in all of them?